weblab/datasets/tests/test_views.py

```diff
@@ -194,6 +194,27 @@ def test_admin_can_delete_dataset(
         assert not dataset.archive_path.exists()
 
 
+@pytest.mark.django_db
+class TestTransferDatasetView:
+
+    def test_datasets_transfer_success(self, client, logged_in_user, other_user, my_dataset_with_file, helpers):
+        dataset = my_dataset_with_file
+        oldpath = dataset.archive_path
+        assert oldpath.exists()
+        assert dataset.author.email == 'test@example.com'
+        response = client.post(
+            '/datasets/%d/transfer' % dataset.pk,
+            data={
+                'email': other_user.email,
+            },
+        )
+        assert response.status_code == 302
+        dataset.refresh_from_db()
+        assert dataset.author == other_user
+        assert not oldpath.exists()
```

Comment:
Based on the bug fix, I suggest we also test `assert not oldpath.parent.exists()`.